apps/google-analytics-4/lambda/src/lib/verify-signed-request-middleware.ts

```diff
@@ -0,0 +1,79 @@
+import * as NodeAppsToolkit from '@contentful/node-apps-toolkit';
+import {
+  CanonicalRequest,
+  ContentfulContextHeader,
+  ContentfulHeader,
+} from '@contentful/node-apps-toolkit/lib/requests/typings';
+import express from 'express';
+import { InvalidSignature } from '../lib/errors/invalid-signature';
+import { UnableToVerifyRequest } from '../lib/errors/unable-to-verify-request';
+import { IncomingHttpHeaders } from 'http';
+
+export const verifySignedRequestMiddleware = (
+  req: express.Request,
+  _res: express.Response,
+  next: express.NextFunction
+) => {
+  const signingSecret = (process.env.SIGNING_SECRET || '').trim();
+  const canonicalReq = makeCanonicalReq(req);
+  let isValidReq = false;
+
+  try {
+    isValidReq = NodeAppsToolkit.verifyRequest(signingSecret, canonicalReq, 60); // 60 second TTL
+  } catch (e) {
+    console.error(e);
+    throw new UnableToVerifyRequest('Unable to verify request', {
+      cause: e,
+    });
+  }
+
+  if (!isValidReq) {
+    throw new InvalidSignature(
+      'Request does not have a valid request signature. ' +
+        'See: https://www.contentful.com/developers/docs/extensibility/app-framework/request-verification/'
+    );
+  }
+
+  next();
+};
+
+const makeCanonicalReq = (req: express.Request) => {
+  const requiredHeaders = requestSigningHeaders(req.headers);
+
+  return <CanonicalRequest>{
```

Comment:
It's not recommended to cast types in TypeScript. I'd set the return value type in line 40. You probably don't get type warnings here.